kokkos.presets.json

```diff
@@ -0,0 +1,68 @@
+{
+    "version"   : 3,
+    "cmakeMinimumRequired" : {
+        "major" : 3,
+        "minor" : 22,
+        "patch" : 0
+    },
+    "configurePresets" : [
+        {
+            "name"          : "default",
+            "displayName"   : "default",
+            "binaryDir"     : "${sourceDir}/build-with-${presetName}",
+            "cacheVariables" : {
+                "CMAKE_EXPORT_COMPILE_COMMANDS" : {"type" : "BOOL"  , "value" : "ON"},
+                "CMAKE_BUILD_TYPE"              : {"type" : "STRING", "value" : "RELEASE"},
+                "CMAKE_CXX_STANDARD"            : {"type" : "STRING", "value" : "17"},
+                "CMAKE_CXX_EXTENSIONS"          : {"type" : "BOOL"  , "value" : "OFF"},
+                "BUILD_SHARED_LIBS"             : {"type" : "BOOL"  , "value" : "ON"}
```

Comment:
Does it matter?

Reply:
I don't know. By default, this is `OFF` in `Kokkos`, see https://github.com/kokkos/kokkos/blob/55bbd9f175307a2ad42d205a3cebb5a32698545b/CMakeLists.txt#L225.

I don't have any opinion on that in this case. If you want, I can remove the `BUILD_SHARED_LIBS` parameter.

Reply:
I think it's a good idea to only set required CMake options for `Kokkos` since that likely matches more closely a usual user configuration.

Reply:
@masterleinad